lib/Serialization/ModuleFile.cpp

```diff
@@ -219,17 +212,7 @@ Status ModuleFile::associateWithFileContext(FileUnit *file, SourceLoc diagLoc) {
       continue;
     }
 
-    if (scopePath.empty()) {
-      dependency.Import =
-          ModuleDecl::ImportedModule{ModuleDecl::AccessPathTy(), module};
-    } else {
-      auto scopeID = ctx.getIdentifier(scopePath);
-      assert(!scopeID.empty() &&
-             "invalid decl name (non-top-level decls not supported)");
-      Located<Identifier> accessPathElem = { scopeID, SourceLoc() };
-      dependency.Import = ModuleDecl::ImportedModule{
-          ctx.AllocateCopy(llvm::makeArrayRef(accessPathElem)), module};
-    }
+    dependency.Import = ModuleDecl::ImportedModule{accessPath, module};
 
```

Comment:
The section between line 190/190 and line 232/215 has been significantly restructured and deserves special attention.

Reply:
@xymus I'd appreciate if you could review at least this portion of the PR (plus the other one I'll ping you in).

Reply:
These changes look good and it's a nice improvement to readability.